modeshape-jcr/src/main/java/org/modeshape/jcr/JcrRepository.java

```diff
@@ -1051,12 +1052,15 @@ protected RunningState( JcrRepository.RunningState other,
                         } else {
                             this.clusteringService = ClusteringService.startStandalone(clusterName, clustering.getConfiguration());        
                         }
+                        JGroupsLockingService jgroupsLockingService = new JGroupsLockingService(this.clusteringService.getChannel(), 
+                                                                                                config.getLockTimeoutMillis());
+                        this.lockingService = clustering.useDbLocking() ? 
+                                              new DbLockingService(jgroupsLockingService, RepositoryCache.INITIALIZATION_LOCK,
+                                                                   this.schematicDb) : jgroupsLockingService;
```

Comment:
Curious why `JGroupsLockingService` instance is always created before knowing whether it's used, while the `DbLockingService` is not. My only naive concern would be whether the `JGroupsLockingService` obtains resources on creation, and if so then it may be doing this unnecessarily.